plugins/single_plugins/idle.cpp

```diff
@@ -329,6 +330,24 @@ class wayfire_idle
         wf::get_core().output_layout->apply_configuration(config);
     }
 
+    void idle_set_enabled(bool enabled)
+    {
+        if (enabled)
+        {
+            idle_ref++;
+
+            if (idle_ref == 1 && idle_enabled)
+                toggle_idle();
+        }
+        else
+        {
+            idle_ref--;
+
```

Comment:
would probably be also nice to print an error in case idle_ref gets below zero since that'd indicate a bug in the plugin.